packages/ckeditor5-html-support/src/datafilter.ts

```diff
@@ -668,6 +699,38 @@ export default class DataFilter extends Plugin {
 			model: attributeKey,
 			view: attributeToViewInlineConverter( definition )
 		} );
+
+		if ( !definition.allowEmpty ) {
+			return;
+		}
+
+		schema.setAttributeProperties( attributeKey, { copyFromObject: false } );
+
+		if ( !schema.isRegistered( 'htmlEmptyElement' ) ) {
+			schema.register( 'htmlEmptyElement', {
+				inheritAllFrom: '$inlineObject'
+			} );
+		}
+
+		editor.data.htmlProcessor.domConverter.registerInlineObjectMatcher( element => (
+			element.name == definition.view && element.isEmpty ? { name: true } : null
+		) );
+		// TODO should it affect spaces vs NBSP in editing pipeline?
```

Comment:
Another approach would be to ignore empty inline elements without any attributes. There still are some side effects but maybe this would be good enough. We simply do not know if the custom CSS class sets the style `display: inline-block` on the given element.
![Screenshot 2023-07-12 at 19 16 31](https://github.com/ckeditor/ckeditor5/assets/1232187/bd95dc3b-a85f-47c9-973c-dcdbe7332edc)